### ISSUE TITLE
Add support for middleware installation

### ISSUE THREAD

**Issue by Mixaill** [gogcom/galaxy-integrations-python-api#73]
It would be nice to have the functionality to request permission from user to install the middleware.

![image](https://user-images.githubusercontent.com/704382/66564622-1b158780-eb69-11e9-88de-15bbb4993a8e.png)

## Api changes

* `plugin.py`
   * add function to call `request_middleware_installation(middlewares: List[Middleware]) -> None`
   * add function for overriding `async def install_middleware(middlewares: List[Middleware]) -> None`
* `types.py`
   * new struct `Middleware`
      * `middleware_id : str` 
      * `middleware_name : str` 

## Execution flow

1) GLX calls `plugin::install_game()`
2) integration checks that some software is not installed.
3) integration calls `plugin::request_middleware_installation()`
4) GLX display window
5) User accept middleware installation
6) GLX calls `plugin::install_middleware()`
7) Integration performs middleware installation

**Comment by rbierbasz-gog** [comment]
Hi - sorry for the late answer. Correct my if I'm wrong, but this feature is basically showing a pop-up in client, isn't it?

**Comment by Mixaill** [comment]
Yeah, something like that. This can be generalized up to the possibility of displaying some information to the user and getting a response to it.
